src/content/docs/en/guides/deploy/gitlab.mdx

```diff
@@ -6,39 +6,90 @@ i18nReady: true
 ---
 import { Steps } from '@astrojs/starlight/components';
 
-You can use [GitLab Pages](https://pages.gitlab.io/) to host an Astro site for your [GitLab](https://about.gitlab.com/) projects, groups, or user account.
+You can use [GitLab Pages](https://docs.gitlab.com/ee/user/project/pages/) to host an Astro site for your [GitLab](https://about.gitlab.com/) projects, groups, or user account.
 
 :::tip[Looking for an example?]
 Check out [the official GitLab Pages Astro example project](https://gitlab.com/pages/astro)!
 :::
 
 ## How to deploy
 
+You can deploy an Astro site to GitLab Pages by using GitLab CI/CD to automatically build and deploy your site. To do this, your source code must be hosted on GitLab and you need to make the following changes to your Astro project:
+
 <Steps>
-1. Set the correct `site` in `astro.config.mjs`.
 
-2. Rename the `public/` directory to `static`.
+1. Set up [`site`](/en/reference/configuration-reference/#site) and [`base`](/en/reference/configuration-reference/#base) options in `astro.config.mjs`.
+
+    ```js title="astro.config.mjs" ins={4-5}
+    import { defineConfig } from 'astro/config';
+
+    export default defineConfig({
+      site: 'https://<username>.gitlab.io',
+      base: '/<my-repo>',
+      outDir: 'public',
+      publicDir: 'static',
+    });
+    ```
+
+    `site`
+
+    The value for `site` must be one of the following:
```

Comment:
These requirements were true specifically for GitHub (these are not requirements for Astro at all, but very specific to GitHub.)

Have you confirmed that this is necessary for GitLab? Is there a reason to assume it works the same way?

Reply:
I have confirmed these requirements by reading the official [GitLab Pages docs](https://docs.gitlab.com/ee/user/project/pages/getting_started_part_one.html):

> You created a project called blog under your username john, therefore your project URL is https://gitlab.com/john/blog/. After you enabled GitLab Pages for this project, and build your site, you can access it at https://john.gitlab.io/blog/.

> You created a group for all your websites called websites, and a project in this group is called blog. Your project URL is https://gitlab.com/websites/blog/. After you enabled GitLab Pages for this project, the site is available at https://websites.gitlab.io/blog/.